### PR TITLE
#35 customized show/hide checkbox video recording

### PR DESCRIPTION
**Pull request by sippsack**
Please have a look at CDATA block in registration.html. How would your solution look like?

### CODE REVIEW

File: src/main/resources/templates/registration.html [hunk 1]
```diff
@@ -54,11 +55,14 @@ <h3>Anmeldung</h3>
         </small>
     </p>
 <script>
-const elems = [];
-function checkPrivacy(elem) {
-    if (elem.checked) elems.push(elem); else elems.pop();
-    document.forms[0].submit.disabled = elems.length !== 2;
-}
+    const elems = [];
+    /*<![CDATA[*/
```
Comment (dasniko) [inline src/main/resources/templates/registration.html:59]: What's that with the CDATA comment?
Something special? Something new? ES-2030?

**Review by dasniko — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/registration.html (hunk 1 above)
Comment (sippsack) [reply src/main/resources/templates/registration.html:59]: I had to initialize a javascript value with a template variable from Quarkus Qute. This results in a syntax error in the IDE. The CDATA block looks like the correct way to solve this problem:

https://stackoverflow.com/questions/25687816/setting-up-a-javascript-variable-from-spring-model-by-using-thymeleaf

Another solution could be parsing a JavaScript string (with the Qute template variable) to an integer:

const numberOfRequiredElements = parseInt("{form.showVideoRecording ? 2 : 1}");

**Review by dasniko — CHANGES_REQUESTED** [review]
Your solution
a) is not complete, as the checkbox for video recording is still visible
b) does not work, even when I  set the "showVideoRecording" to false, I have to check both checkboxes
c) should provide a better refactoring of the "checkPrivacy" function, IMHO. This function was a quick hack to provide some logic, but if we now need some more sophisticated logic, we should also refactor the function and check the checkboxes itself, not only the count. This way, we are able to avoid the CDATA element.

**Comment by dasniko** [comment]
If you need help, let me know.

**Comment by dasniko** [comment]
And BTW, naming the switch „hideVideoRecording“ with a default value of „false“ would be better/easier to understand.

**Comment by sippsack** [comment]
Thanks for your input. Weird, that the conditional block disappeared. Will have a look and try it myself. Have already set up the local dev environment. Looks good so far.

**Review by dasniko — APPROVED** [review]

File: README.adoc [hunk 2]
```diff
@@ -59,6 +63,16 @@ The container will start and initialize all the needed AWS resoures like DynamoD
...
++
+[IMPORTANT]
+====
+If an error occurs: "software.amazon.awssdk.core.exception.SdkClientException: Unable to load credentials from any of the providers in the chain", you need to set following env variables:
+
+export AWS_ACCESS_KEY_ID=foobar
+
+export AWS_SECRET_ACCESS_KEY=foobar
+====
+
```
Comment (dasniko) [inline README.adoc:75]: Next time, please don't mix PRs with different issues. File a new PR instead. Thanks!

**Pull request merged**